### ISSUE TITLE
Broken lighting in TR5

### ISSUE THREAD

**Issue by vvs-** [opentomb/OpenTomb#46]
But the lighting problems are inconsistent. I get them in the same places but not every time. Seems that they depend on previous actions. Every time I load the level I get different distribution. They appear as very bright white, yellow or totally black colors on Lara and some textures. Here's one place where it sometimes can be seen: `setgamef(5,1)` `setEntityPos(player,46970,42791,-256,270,0,0)`. And here's another: `setEntityPos(player,51278,38426,-256,90,0,0)`.

I'm not able to attach screenshots for some reason, so here's the [link](https://gist.github.com/vvs-/ae76d1958bb0dfa78488) to them.

**Comment by vvs-** [comment]
Yeah, there is no hyper saturation anymore, but lighting is still inconsistent each time.

**Comment by Lwmte** [comment]
There were lighting problems long before Cochrane's shader conversion, actually it was never fixed by previous person who worked on lighting before (it was Saracen). So I suppose that TR5 lighting problems could not be related to these particular commits.

**Comment by cochrane** [comment]
Yeah, I'm fairly certain they're not. I looked at the values and got an intensity of -40000 at one point. The problems are earlier in the chain.

Keeping this bug open to remember to solve this later.

**Comment by Gh0stBlade** [comment]
I looked at this a long time ago IIRC the problem is the TR5 light struct is being read incorrectly, most likely data type mis-match.
